### PR TITLE
EOS-13526 beck.c: filter bnodes based on generation identifier in beck tool

### PR DESCRIPTION
**Pull request by NikhilBirgade**
Following cases are covered:

1) If source segment header generation identifier is zero then process all the
bnodes.
2) If source segment header generation identifer is non-zero then process only
the bnodes with matching generation identifier.

Signed-off-by: Nikhil Kumar Birgade <nikhil.birgade@seagate.com>

### CODE REVIEW

File: be/tool/beck.c [hunk 1]
```diff
@@ -1107,13 +1125,19 @@ static int seghdr_ver(struct scanner *s, struct rectype *r, char *buf)
...
 	int i;
+	if (gen_count >= ARRAY_SIZE(g))
+		return;
 
-	for (i = 0; i < ARRAY_SIZE(g); ++i) {
-		if (g[i].g_gen == gen || g[i].g_count == 0) {
+	for (i = 0; i < gen_count; ++i) {
+		if (g[i].g_gen == gen) {
 			g[i].g_count++;
 			break;
```
Comment (huanghua78) [inline be/tool/beck.c:1134]: Here, it can just be "return;"

File: be/tool/beck.c [hunk 2]
```diff
@@ -1107,13 +1125,19 @@ static int seghdr_ver(struct scanner *s, struct rectype *r, char *buf)
...
 		}
 	}
+	if (i == gen_count) {
+		g[gen_count].g_gen = gen;
+		g[gen_count++].g_count = 1;
```
Comment (huanghua78) [inline be/tool/beck.c:1139]: It's better not to use the self increment operation here:
It is error prone. 

Let's use a separate clause:
```
g[gen_count].g_count = 1;
gen_count++;

```

File: be/tool/beck.c (hunk 1 above)
Comment (NikhilBirgade) [reply be/tool/beck.c:1134]: Done.

File: be/tool/beck.c (hunk 2 above)
Comment (NikhilBirgade) [reply be/tool/beck.c:1139]: Done.

**Review by RakeshVaghasiya — APPROVED** [review]

**Review by shashank-parulekar — APPROVED** [review]

**Review by madhavemuri — APPROVED** [review]

File: be/tool/beck.c [hunk 3]
```diff
@@ -1108,12 +1131,18 @@ static void genadd(uint64_t gen)
...
+	for (i = 0; i < gen_count; ++i) {
+		if (g[i].g_gen == gen) {
 			g[i].g_count++;
-			break;
+			return;
 		}
 	}
+	g[gen_count].g_gen = gen;
+	g[gen_count].g_count = 1;
+	gen_count++;
```
Comment (nikitadanilov) [inline be/tool/beck.c:1145]: What was wrong with the original code?

**Pull request merged**